cloudinit/net/__init__.py

```diff
@@ -765,7 +765,7 @@ def find_entry(mac, driver, device_id):
     for mac, new_name, driver, device_id in renames:
         if mac:
             mac = mac.lower()
-        cur_ops = []
+        cur_ops: list = []
```

Comment:
on <3.9, lowercase technically works if you're not subscripting it, but I think we should stick with upper case until we move to 3.9.

Reply:
> but I think we should stick with upper case until we move to 3.9.

Why? What we would gain by preferring the deprecated version? This is forwards compatible and cloud-init's codebase uses `list` and `dict` extensively in type annotations where not subscripted.

Reply:
> What we would gain by preferring the deprecated version?

`List` and `list` are seen differently by mypy in < 3.8. `List` is a generic type whereas `list` is not. Even if you're not parametrizing it, I'm pretty sure mypy treats them differently (i.e., if you've ever had mypy yell at you to provide parametrize a generic type).

That said, since we're running mypy in CI against a higher python version, I suppose it'll get raised in CI even if it isn't shown locally so this is fine.